cmd/juju/application/bundle.go

```diff
@@ -1039,6 +1063,23 @@ func (h *bundleHandler) setAnnotations(change *bundlechanges.SetAnnotationsChang
 	return nil
 }
 
+// createOffer creates an offer targeting one or more application endpoints.
+func (h *bundleHandler) createOffer(change *bundlechanges.CreateOfferChange) error {
+	if h.dryRun {
```

Comment:
I'm never sure if this actually does enough to actually facility entirely a "dry run"... would be interested to work that out...

Reply:
I guess any change that fires API calls should be suppressed in dry-run mode to avoid getting errors about missing/not deployed things.